storage/remote/storage.go

```diff
@@ -62,42 +63,53 @@ func (s *Storage) ApplyConfig(conf *config.Config) error {
 
 	// Update write queues
 
-	newQueues := []*QueueManager{}
-	// TODO: we should only stop & recreate queues which have changes,
-	// as this can be quite disruptive.
+	newQueues := make([]*QueueManager, 0, len(s.queues))
```

Comment:
wouldn't `len(conf.RemoteWriteConfigs)` make more sense here for the new length?